src/coreclr/src/tools/crossgen2/ILCompiler.ReadyToRun/Compiler/ReadyToRunCompilationModuleGroupBase.cs

```diff
@@ -220,6 +223,8 @@ public sealed override bool TryGetModuleTokenForExternalType(TypeDesc type, out
 
         public sealed override bool IsCompositeBuildMode => _isCompositeBuildMode;
 
+        public sealed override bool IsLibraryBuildMode => _isLibraryBuildMode;
```

Comment:
Is the `IsLibraryBuildMode` flag used for anything? I am not able to find it.

Reply:
Hmm, the idea was that the library build mode can be used e.g. to tune rooting heuristics. You're right it's not wired up this way in this iteration; I can easily revert it if you don't think it's of any use.

Reply:
OK, for now I'm going to remove this as speculative code is generally not desirable and the current compilation logic doesn't actually need this.

Reply:
This can be easily added once it is actually needed. I do not think we should be adding this without the actual use.

Reply:
Removed in 4th commit.